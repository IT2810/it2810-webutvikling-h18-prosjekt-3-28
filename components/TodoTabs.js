import React from 'react';
import {
  Image,
  Platform,
  ScrollView,
  StyleSheet,
  TouchableOpacity,
  View,
  Alert,
} from 'react-native';
import {
  ButtonGroup,
} from 'react-native-elements';

import{
  LinearGradient,
} from 'expo';


class TodoTabs extends React.Component{

  constructor () {
    super()
    this.state = {
      selectedIndex: 0
    }
    this.updateIndex = this.updateIndex.bind(this)
  }
  
  updateIndex (selectedIndex) {
    this.setState({selectedIndex})
  }
  
  render () {
    const buttons = ['ToDos', 'Appointments',]
    const { selectedIndex } = this.state
  
    return (    
      <ButtonGroup
      onPress={this.updateIndex}
      selectedIndex={selectedIndex}
      buttons={buttons}
      selectedButtonStyle={{backgroundColor:"#0080ff"}}
      selectedTextStyle={{color:"#ffffff"}}
      containerStyle={{height: 40}}
    />
    )
  }
} 

export default TodoTabs