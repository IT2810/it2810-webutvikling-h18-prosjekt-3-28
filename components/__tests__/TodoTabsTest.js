import React from 'react';
import TodoTabs from '../TodoTabs';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';

describe('App snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
      NavigationTestUtils.resetInternalState();
    });
  
    it('renders the loading screen', async () => {
      const tree = renderer.create(<TodoTabs />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('renders the root without loading screen', async () => {
      const tree = renderer.create(<TodoTabs skipLoadingScreen />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});

