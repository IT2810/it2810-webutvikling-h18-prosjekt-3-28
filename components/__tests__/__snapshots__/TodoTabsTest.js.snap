// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the loading screen 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e3e3e3",
        "borderRadius": 3,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 5,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 5,
        "overflow": "hidden",
      },
      Object {
        "height": 40,
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": 3,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 0,
        "borderTopLeftRadius": 3,
        "flex": 1,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
          Object {
            "backgroundColor": "#2089dc",
          },
          Object {
            "backgroundColor": "#0080ff",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#5e6977",
                "fontSize": 16.25,
                "fontWeight": "500",
              },
              undefined,
              Object {
                "color": "#fff",
              },
              Object {
                "color": "#ffffff",
              },
            ],
          ]
        }
      >
        ToDos
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomRightRadius": 3,
        "borderLeftColor": "#bdc6cf",
        "borderLeftWidth": 1,
        "borderTopRightRadius": 3,
        "flex": 1,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
          false,
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#5e6977",
                "fontSize": 16.25,
                "fontWeight": "500",
              },
              undefined,
              false,
              false,
            ],
          ]
        }
      >
        Appointments
      </Text>
    </View>
  </View>
</View>
`;

exports[`App snapshot renders the root without loading screen 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e3e3e3",
        "borderRadius": 3,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 5,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 5,
        "overflow": "hidden",
      },
      Object {
        "height": 40,
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": 3,
        "borderRightColor": "#bdc6cf",
        "borderRightWidth": 0,
        "borderTopLeftRadius": 3,
        "flex": 1,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
          Object {
            "backgroundColor": "#2089dc",
          },
          Object {
            "backgroundColor": "#0080ff",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#5e6977",
                "fontSize": 16.25,
                "fontWeight": "500",
              },
              undefined,
              Object {
                "color": "#fff",
              },
              Object {
                "color": "#ffffff",
              },
            ],
          ]
        }
      >
        ToDos
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomRightRadius": 3,
        "borderLeftColor": "#bdc6cf",
        "borderLeftWidth": 1,
        "borderTopRightRadius": 3,
        "flex": 1,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
          false,
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#5e6977",
                "fontSize": 16.25,
                "fontWeight": "500",
              },
              undefined,
              false,
              false,
            ],
          ]
        }
      >
        Appointments
      </Text>
    </View>
  </View>
</View>
`;
